### PR TITLE
[Issue-2] Added authentication logic and login flow

### PR DESCRIPTION
**Pull request by michaelbukachi**
## What is the Purpose?
Added authentication logic and login flow

## What was the approach?
Added logic to handle authentication and token management

## Are there any concerns to addressed further before or after merging this PR?
The backend needs to be running locally in order to go through the authentication process.

## Mentions?
@andrewtpham @ninetteadhikari @nasirhjafri 

## Issue(s) affected?
#2

### CODE REVIEW

File: src/auth.js [hunk 1]
```diff
@@ -0,0 +1,60 @@
...
+      next({ path: '/' })
+      NProgress.done()
+    } else {
+      const nameLoaded = !!store.getters.name
+
+      if (nameLoaded) {
+        next()
+      } else {
+        try {
+          await store.dispatch('user/getInfo')
```
Comment (nasirhjafri) [inline src/auth.js:34]: Will this send a request every time page is refreshed or every time route is changed?

File: src/store/modules/user.js [hunk 2]
```diff
@@ -0,0 +1,78 @@
...
+      })
+    })
+  },
+
+  getInfo({ commit, state }) {
+    return new Promise((resolve, reject) => {
+      getInfo().then(response => {
+        const { email, fullName } = response
+
+        if (!email || !fullName) {
```
Comment (nasirhjafri) [inline src/store/modules/user.js:42]: Let's remove this part.

File: src/store/modules/user.js [hunk 3]
```diff
@@ -0,0 +1,78 @@
...
+          reject('Verification failed, please Login again.')
+        }
+
+        commit('SET_EMAIL', email)
```
Comment (nasirhjafri) [inline src/store/modules/user.js:46]:
```
commit('USER',  user)
```

File: src/utils/request.js [hunk 4]
```diff
@@ -0,0 +1,55 @@
...
+  error => {
+    console.error(error)
+    return Promise.reject(error)
+  }
+)
+
+service.interceptors.response.use(
+  response => {
+    if (response.status !== 200 && response.status !== 201) {
+      if (response.status === 401) {
```
Comment (nasirhjafri) [inline src/utils/request.js:39]: We will not be logging out the user instead we will do a request to refresh the token.
401 on any endpoint, do a request to refresh token.
If 200 on refresh token request, update the access token and return the original request again.
If 401 on refresh token request, logout the user.

File: src/auth.js (hunk 1 above)
Comment (michaelbukachi) [reply src/auth.js:34]: When the page is refreshed.

Comment (nasirhjafri) [reply src/auth.js:34]: hmm cool, I don't want it to be triggered every time route changes i.e. a different component is rendered.

**Comment by andrewtpham** [comment]
@michaelbukachi let me know when this is ready to review again, also seeing conflicts in the branch.

**Comment by michaelbukachi** [comment]
@andrewtpham It's ready for review.

**Comment by nasirhjafri** [comment]
@michaelbukachi Its not refreshing the token when the access token is expired.

**Pull request merged**